### DISCUSSION TITLE
Is a personal access token needed?

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#26998] [Actions]
<p>I have the following Action set up in order to push files to the GitHub Pages branch using MkDocs.</p>
<p>The question I now have is, if the GH_TOKEN, which is a Personal Access Token, is required for this to work, or if I can replace it with something else (or remove it altogether).</p>
<p>The YAML file I have setup</p>
<pre><code>name: Deploy Site

on:
  push:
    paths:
    - 'docs/**'
    - '**.yml'
    - 'theme/**'
    branches:
    - master

jobs:
  build:
    runs-on: [ubuntu-latest]
    steps:
    - uses: actions/checkout@v2
    - name: Set up Python 3.7
      uses: actions/setup-python@v1
      with:
        python-version: 3.7
    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip setuptools
        python -m pip install -r requirements.txt # File only has mkdocs-material &gt;= 5.0.0
    - name: Deploy Files
      run: |
        git config user.name "github-actions[bot]" # I use the GitHub Actions bot here.
        git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
        git remote add gh-token "https://${{ secrets.GH_TOKEN}}@github.com/Andre601/MyRepo.git" # Example
        git fetch gh-token &amp;&amp; git fetch gh-token gh-pages:gh-pages
        python -m mkdocs gh-deploy --clean --remote-name gh-token
        git push gh-token gh-pages
</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/andre601">@andre601</a> ,</p>
<p>If you’re pushing commit to ‘gh-pages’ branch in same repository, you can skip token:</p>
<pre><code>git remote add gh-token "https://github.com/Andre601/MyRepo.git"
</code></pre>
<p>This is due to ‘actions/checkout’ will automatically add <strong>GITHUB_TOKEN</strong> to later git operations, and the token has write permission to current repo.</p>
<p>If you push commit to another repository, personal access token is needed. Code sample as below:</p>
<pre><code>git remote add gh-token "https://weide-zhou:${{ secrets.PAT1 }}@github.com/weide-zhou/ticket14.git"
</code></pre>
<p>Thanks.</p>

**Comment by Andre601** [comment]
<p>Thanks.</p>
<p>It works without any issues.</p>
